src/main/scala/pipedsl/typechecker/LockConstraintChecker.scala

```diff
@@ -166,7 +166,7 @@ class LockConstraintChecker(lockMap: Map[Id, Set[LockArg]], lockGranularityMap:
           } else {
             env
           }
-        case (_, ECall(_, _, args)) =>
+        case (_, ECall(_, _, args, _)) =>
```

Comment:
We definitely need to check to see if this call has an associated lock reserved (if it's not atomic) - that's the main use case for reservations.